### PR TITLE
BZ2011298: Missing information wrt cloud formation template in AWS installation of Openshift

### PR DESCRIPTION
**Pull request by xenolinux**
Fixes: [BZ2011298](https://bugzilla.redhat.com/show_bug.cgi?id=2011298)
OCP version: 4.6+
QE contact: @yunjiang29 
[preview](https://deploy-preview-40854--osdocs.netlify.app/openshift-enterprise/latest/installing/installing_aws/installing-restricted-networks-aws.html#installation-aws-user-infra-other-infrastructure_installing-restricted-networks-aws)

### CODE REVIEW

**Comment by yunjiang29** [comment]
per https://bugzilla.redhat.com/show_bug.cgi?id=2011298#c0, looks like following ports are missing:
```
- 9000 - 9999/udp
- 30000 - 32767/udp
```
which were introduced by [Bug 1779469](https://bugzilla.redhat.com/show_bug.cgi?id=1779469) - Regression: [sig-network] Networking Granular Checks: Services should function for endpoint-Service: udp [Suite:openshift/conformance/parallel] [Suite:k8s] [Skipped:Network/OVNKubernetes]

@staebler @andreaskaris PTAL, thanks

**Comment by staebler** [comment]
> per https://bugzilla.redhat.com/show_bug.cgi?id=2011298#c0, looks like following ports are missing:
> 
> ```
> - 9000 - 9999/udp
> - 30000 - 32767/udp
> ```

Yes, those ports are used for ingress, which can use UDP.

File: modules/installation-aws-user-infra-requirements.adoc [hunk 1]
```diff
@@ -290,6 +290,31 @@ a `AWS::EC2::SecurityGroupIngress` resource.
...
+|`udp`
+|`500`
+
+|`MasterIngressIpsecNat`
+|ovn-kubernetes IPsec
+|`udp`
+|`4500`
+
+|`MasterIngressInternal`
+|ovn-kubernetes IPsec
```
Comment (yunjiang29) [inline modules/installation-aws-user-infra-requirements.adoc:309]: should be `Internal cluster communication`

File: modules/installation-aws-user-infra-requirements.adoc [hunk 2]
```diff
@@ -290,6 +290,31 @@ a `AWS::EC2::SecurityGroupIngress` resource.
...
+|`udp`
+|`9000` - `9999`
+
+|`MasterIngressIngressServices`
+|ovn-kubernetes IPsec
```
Comment (yunjiang29) [inline modules/installation-aws-user-infra-requirements.adoc:314]: should be `Internal cluster communication`

File: modules/installation-aws-user-infra-requirements.adoc [hunk 3]
```diff
@@ -347,6 +372,31 @@ a `AWS::EC2::SecurityGroupIngress` resource.
...
+|`udp`
+|`500`
+
+|`MasterIngressWorkerIpsecNat`
+|ovn-kubernetes IPsec
+|`udp`
+|`4500`
+
+|`MasterIngressWorkerInternal`
+|ovn-kubernetes IPsec
```
Comment (yunjiang29) [inline modules/installation-aws-user-infra-requirements.adoc:391]: should be `Internal cluster communication`

File: modules/installation-aws-user-infra-requirements.adoc [hunk 4]
```diff
@@ -347,6 +372,31 @@ a `AWS::EC2::SecurityGroupIngress` resource.
...
+|`udp`
+|`9000` - `9999`
+
+|`MasterIngressWorkerIngressServices`
+|ovn-kubernetes IPsec
```
Comment (yunjiang29) [inline modules/installation-aws-user-infra-requirements.adoc:396]: should be `Internal cluster communication`

Comment (xenolinux) [reply modules/installation-aws-user-infra-requirements.adoc]: Addressed

File: modules/installation-aws-user-infra-requirements.adoc (hunk 2 above)
Comment (xenolinux) [reply modules/installation-aws-user-infra-requirements.adoc]: Addressed

**Comment by xenolinux** [comment]
@yunjiang29 I addressed the changes can you check once if it looks good?

**Comment by yunjiang29** [comment]
@xenolinux I strongly recommend checking CloudFormation template carefully, as we need to fill the gap between CF template and documentation for particular protocol and ports, this is what is mentioned in the bug, thanks. 

The CloudFormation templates are available in both [document](https://docs.openshift.com/container-platform/4.8/installing/installing_aws/installing-restricted-networks-aws.html#installation-aws-user-infra-other-infrastructure_installing-restricted-networks-aws
) and [installer repo](https://github.com/yunjiang29/installer/blob/release-4.8/upi/aws/cloudformation/03_cluster_security.yaml).

The contents that I found in the following table exist in the CF templates but missing in the document, for your reference.
(Please note that the group names for `Control plane Ingress` and `Worker Ingress` are similar but different, I saw you put all `Master*` groups in the `Worker Ingress` table.)

## Control plane Ingress

| Ingress group  | Description | IP protocol  | Port range |
| ------------- | ------------- | ------------- | ------------- |
| `MasterIngressGeneve`  | Geneve packets  | `udp`  | `6081`  |
| `MasterIngressWorkerGeneve`  | Geneve packets  | `udp`  | `6081`  |
| `MasterIngressIpsecIke`  | IPsec IKE packets  | `udp`  | `500`  |
| `MasterIngressWorkerIpsecIke`  | IPsec IKE packets  | `udp`  | `500`  |
| `MasterIngressIpsecNat`  | IPsec NAT-T packets  | `udp`  | `4500`  |
| `MasterIngressWorkerIpsecNat`  | IPsec NAT-T packets  | `udp`  | `4500`  |
| `MasterIngressIpsecEsp`  | IPsec ESP packets  | `50`  | `-`  |
| `MasterIngressWorkerIpsecEsp`  | IPsec ESP packets  | `50`  | `-`  |
| `MasterIngressInternalUDP`  | Internal cluster communication  | `udp`  | `9000` - `9999`  |
| `MasterIngressWorkerInternalUDP`  | Internal cluster communication  | `udp`  | `9000` - `9999`  |
| `MasterIngressIngressServicesUDP`  | Kubernetes ingress services  | `udp`  | `30000` - `32767`  |
| `MasterIngressWorkerIngressServicesUDP`  | Kubernetes ingress services  | `udp`  | `30000` - `32767`  |


## Worker Ingress

| Ingress group  | Description | IP protocol  | Port range |
| ------------- | ------------- | ------------- | ------------- |
| `WorkerIngressGeneve`  | Geneve packets  | `udp`  | `6081`  |
| `WorkerIngressMasterGeneve`  | Geneve packets  | `udp`  | `6081`  |
| `WorkerIngressIpsecIke`  | IPsec IKE packets  | `udp`  | `500`  |
| `WorkerIngressMasterIpsecIke`  | IPsec IKE packets  | `udp`  | `500`  |
| `WorkerIngressIpsecNat`  | IPsec NAT-T packets  | `udp`  | `4500`  |
| `WorkerIngressMasterIpsecNat`  | IPsec NAT-T packets  | `udp`  | `4500`  |
| `WorkerIngressIpsecEsp`  | IPsec ESP packets  | `50`  | `-`  |
| `WorkerIngressMasterIpsecEsp`  | IPsec ESP packets  | `50`  | `-`  |
| `WorkerIngressInternalUDP`  | Internal cluster communication  | `udp`  | `9000` - `9999`  |
| `WorkerIngressMasterInternalUDP`  | Internal cluster communication  | `udp`  | `9000` - `9999`  |
| `WorkerIngressIngressServicesUDP`  | Kubernetes ingress services  | `udp`  | `30000` - `32767`  |
| `WorkerIngressMasterIngressServicesUDP`  | Kubernetes ingress services  | `udp`  | `30000` - `32767`  |

**Comment by yunjiang29** [comment]
A minor update for https://github.com/openshift/openshift-docs/pull/40854#issuecomment-1041375163: Changed `Port range`  from `-` to `All` for Protocol 50

File: modules/installation-aws-user-infra-requirements.adoc [hunk 5]
```diff
@@ -290,6 +290,66 @@ a `AWS::EC2::SecurityGroupIngress` resource.
...
+|`udp`
+|`9000` - `9999`
+
+|`MasterIngressWorkerInternalUDP`
+|Internal cluster communication
+|`udp`
+|`9000` - `9999`
+
+|`MasterIngressIngressServicesUDP`
+|Kubernetes ingress services
```
Comment (yunjiang29) [inline modules/installation-aws-user-infra-requirements.adoc:344]: To keep the name consistent, change `ingress` to `Ingress`.

File: modules/installation-aws-user-infra-requirements.adoc [hunk 6]
```diff
@@ -290,6 +290,66 @@ a `AWS::EC2::SecurityGroupIngress` resource.
...
+|`udp`
+|`30000` - `32767`
+
+|`MasterIngressWorkerIngressServicesUDP`
+|Kubernetes ingress services
```
Comment (yunjiang29) [inline modules/installation-aws-user-infra-requirements.adoc:349]: To keep the name consistent, change `ingress` to `Ingress`.

File: modules/installation-aws-user-infra-requirements.adoc [hunk 7]
```diff
@@ -347,6 +407,66 @@ a `AWS::EC2::SecurityGroupIngress` resource.
...
+|`udp`
+|`9000` - `9999`
+
+|`WorkerIngressIngressServicesUDP`
+|Kubernetes ingress services
+|`udp`
+|`30000` - `32767`
+
+|`WorkerIngressMasterIngressServicesUDP`
+|Kubernetes ingress services
```
Comment (yunjiang29) [inline modules/installation-aws-user-infra-requirements.adoc:466]: To keep the name consistent, change `ingress` to `Ingress`.

File: modules/installation-aws-user-infra-requirements.adoc [hunk 8]
```diff
@@ -347,6 +407,66 @@ a `AWS::EC2::SecurityGroupIngress` resource.
...
+|`udp`
+|`9000` - `9999`
+
+|`WorkerIngressMasterInternalUDP`
+|Internal cluster communication
+|`udp`
+|`9000` - `9999`
+
+|`WorkerIngressIngressServicesUDP`
+|Kubernetes ingress services
```
Comment (yunjiang29) [inline modules/installation-aws-user-infra-requirements.adoc:461]: To keep the name consistent, change `ingress` to `Ingress`.

File: modules/installation-aws-user-infra-requirements.adoc [hunk 9]
```diff
@@ -347,6 +407,66 @@ a `AWS::EC2::SecurityGroupIngress` resource.
...
+|`4500`
+
+|`WorkerIngressMasterIpsecNat`
+|IPsec NAT-T packets
+|`udp`
+|`4500`
+
+|`WorkerIngressIpsecEsp`
+|IPsec ESP packets
+|`-`
```
Comment (yunjiang29) [inline modules/installation-aws-user-infra-requirements.adoc:442]: the protocol name is `50`

File: modules/installation-aws-user-infra-requirements.adoc [hunk 10]
```diff
@@ -347,6 +407,66 @@ a `AWS::EC2::SecurityGroupIngress` resource.
...
+|`All`
+
+|`WorkerIngressMasterIpsecEsp`
+|IPsec ESP packets
+|`-`
```
Comment (yunjiang29) [inline modules/installation-aws-user-infra-requirements.adoc:447]: the protocol name is `50`

File: modules/installation-aws-user-infra-requirements.adoc [hunk 11]
```diff
@@ -290,6 +290,66 @@ a `AWS::EC2::SecurityGroupIngress` resource.
...
+|`4500`
+
+|`MasterIngressIpsecEsp`
+|IPsec ESP packets
+|`-`
+|`All`
+
+|`MasterIngressWorkerIpsecEsp`
+|IPsec ESP packets
+|`-`
```
Comment (yunjiang29) [inline modules/installation-aws-user-infra-requirements.adoc:330]: the protocol name is `50`

File: modules/installation-aws-user-infra-requirements.adoc [hunk 12]
```diff
@@ -290,6 +290,66 @@ a `AWS::EC2::SecurityGroupIngress` resource.
...
+|`4500`
+
+|`MasterIngressWorkerIpsecNat`
+|IPsec NAT-T packets
+|`udp`
+|`4500`
+
+|`MasterIngressIpsecEsp`
+|IPsec ESP packets
+|`-`
```
Comment (yunjiang29) [inline modules/installation-aws-user-infra-requirements.adoc:325]: the protocol name is `50`

Comment (xenolinux) [reply modules/installation-aws-user-infra-requirements.adoc]: Addressed

File: modules/installation-aws-user-infra-requirements.adoc (hunk 11 above)
Comment (xenolinux) [reply modules/installation-aws-user-infra-requirements.adoc]: Addressed

File: modules/installation-aws-user-infra-requirements.adoc (hunk 10 above)
Comment (xenolinux) [reply modules/installation-aws-user-infra-requirements.adoc]: Addressed

File: modules/installation-aws-user-infra-requirements.adoc (hunk 9 above)
Comment (xenolinux) [reply modules/installation-aws-user-infra-requirements.adoc]: Addressed

File: modules/installation-aws-user-infra-requirements.adoc (hunk 8 above)
Comment (xenolinux) [reply modules/installation-aws-user-infra-requirements.adoc]: Addressed

File: modules/installation-aws-user-infra-requirements.adoc (hunk 7 above)
Comment (xenolinux) [reply modules/installation-aws-user-infra-requirements.adoc]: Addressed

File: modules/installation-aws-user-infra-requirements.adoc (hunk 6 above)
Comment (xenolinux) [reply modules/installation-aws-user-infra-requirements.adoc]: Addressed

File: modules/installation-aws-user-infra-requirements.adoc (hunk 5 above)
Comment (xenolinux) [reply modules/installation-aws-user-infra-requirements.adoc]: Addressed

**Comment by bergerhoffer** [comment]
Kicking off the cherry picks for 4.6-4.11

**Pull request merged**